### PR TITLE
Implement accumulator updates for cosmwasm

### PR DESCRIPTION
**Pull request by m30m**

### CODE REVIEW

**Review by jayantk — APPROVED** [review]
looks great!

File: target_chains/cosmwasm/contracts/pyth/src/contract.rs [hunk 1]
```diff
@@ -1003,6 +1106,333 @@ mod test {
...
+        let mut feed1 = create_dummy_price_feed_message(100);
+        let mut feed2 = create_dummy_price_feed_message(200);
+        let mut feed3 = create_dummy_price_feed_message(300);
+        let msg = create_accumulator_message(&[feed1, feed2, feed3], &[feed1, feed2, feed3], false);
+        to_price_feed(&mut feed1).publish_time += 1;
+        to_price_feed(&mut feed2).publish_time += 1;
+        to_price_feed(&mut feed3).publish_time += 1;
+        to_price_feed(&mut feed1).price *= 2;
+        to_price_feed(&mut feed2).price *= 2;
+        to_price_feed(&mut feed3).price *= 2;
```
[the remark this answers is not in the document]
Comment (m30m) [reply target_chains/cosmwasm/contracts/pyth/src/contract.rs]: Tried to conform to [this](https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv)

**Review by ali-bahjati — APPROVED** [review]
Very nice!

I think it would be good to add a few more tests that i mentioned in an inline comment. Please add them before merging.

**Review by ali-bahjati — APPROVED** [review]
🚢 it

**Pull request merged**